### PR TITLE
Ags commit

### PR DESCRIPTION
**Pull request by brmscheiner**

### CODE REVIEW

File: src/pages/sighting/AssetGroupSighting.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  const {
+    data: assetGroupSightingData,
+    loading: assetGroupLoading,
+    error: assetGroupError,
+    statusCode: assetGroupStatusCode,
+  } = useAssetGroupSighting(id);
+
+  const sightingId = assetGroupSightingData?.sightingGuid;
+
+  useEffect(
```
Comment (brmscheiner) [inline src/pages/sighting/AssetGroupSighting.jsx:20]: automatically redirect to sightings page if AGS has already been turned into a sighting

File: src/pages/sighting/SightingCore.jsx [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+// import AnnotationsGallery from './AnnotationsGallery';
+// import IndividualsGallery from './IndividualsGallery';
+import Annotations from './Annotations';
+import Photographs from './Photographs';
+import OverviewContent from './OverviewContent';
+import SightingHistoryDialog from './SightingHistoryDialog';
+import FeaturedPhoto from './featuredPhoto/FeaturedPhoto';
+import Encounters from './encounters/Encounters';
+
+export default function SightingCore({
```
Comment (brmscheiner) [inline src/pages/sighting/SightingCore.jsx:44]: note that not all this code is new - most of it is moved from the old SightingCore component

File: src/pages/sighting/AssetGroupSighting.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+export default function AssetGroupSighting() {
+  const history = useHistory();
+  const { id } = useParams();
+
+  const {
+    data: assetGroupSightingData,
+    loading: assetGroupLoading,
+    error: assetGroupError,
+    statusCode: assetGroupStatusCode,
```
Comment (Atticus29) [inline src/pages/sighting/AssetGroupSighting.jsx:15]: I'm not seeing statusCode in the react-query documentation nor in useAssetGroupSighting.js... where is this coming from?

File: src/pages/sighting/AssetGroupSighting.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  } = useAssetGroupSighting(id);
+
+  const sightingId = assetGroupSightingData?.sightingGuid;
+
+  useEffect(
+    () => {
+      if (sightingId) {
+        history.push(`/sightings/${sightingId}`);
```
Comment (Atticus29) [inline src/pages/sighting/AssetGroupSighting.jsx:23]: I'm a little confused. Does this actually do the redirect or just push it into an array that we can check later and navigate away based on presence/absence in history?

Comment (Atticus29) [reply src/pages/sighting/AssetGroupSighting.jsx:23]: If the latter, where is the logic that is actually changing the route?

File: src/pages/sighting/Sighting.jsx [hunk 5]
```diff
@@ -1,270 +1,27 @@
...
-import Link from '../../components/Link';
-import Text from '../../components/Text';
-import LoadingScreen from '../../components/LoadingScreen';
-import SadScreen from '../../components/SadScreen';
-import Button from '../../components/Button';
-import MoreMenu from '../../components/MoreMenu';
-import ConfirmDelete from '../../components/ConfirmDelete';
-import EntityHeaderNew from '../../components/EntityHeaderNew';
-import useAssetGroupSighting from '../../models/assetGroup/useAssetGroupSighting';
+import SightingCore from './SightingCore';
```
Comment (Atticus29) [inline src/pages/sighting/Sighting.jsx:4]: This now seems like  a wrapper for SightingCore. Why is this file necessary?

File: src/pages/sighting/SightingCore.jsx [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+  const {
+    commitAgs,
+    isLoading: commitAgsLoading,
+  } = useCommitAssetGroupSighting();
+
+  /*
+    known issue: if data or fieldschemas change values
+    or properties this may not update
+    switch to data.version?
+  */
```
Comment (Atticus29) [inline src/pages/sighting/SightingCore.jsx:81]: Hm. I wonder why.

File: src/pages/sighting/SightingCore.jsx [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+  if (loading) return <LoadingScreen />;
+  if (statusCode === 404)
+    return (
+      <SadScreen
+        subtitleId="SIGHTING_NOT_FOUND"
+        descriptionId="SIGHTING_NOT_FOUND_DESCRIPTION"
+        variant="genericError"
+      />
+    );
+  if (error) return <SadScreen variant="genericError" />;
```
Comment (Atticus29) [inline src/pages/sighting/SightingCore.jsx:110]: This feels like the least specific conditional. Should this not be the last one encountered? In other words, why not move this below the if(!data) section below?

File: src/pages/sighting/SightingCore.jsx [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+        variant="notFoundOcean"
+        subtitleId="SIGHTING_NOT_FOUND"
+      />
+    );
+
+  const assets = get(data, 'assets', []);
+  const annotations = flatten(
+    assets.map(asset => get(asset, 'annotations')),
+  );
+  const allAnnotationsAssigned = every(annotations, 'encounter_guid');
```
Comment (Atticus29) [inline src/pages/sighting/SightingCore.jsx:123]: TIL. Cool.

File: src/pages/sighting/SightingCore.jsx [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+  const detectionComplete = get(data, 'stage') !== 'detection';
```
Comment (Atticus29) [inline src/pages/sighting/SightingCore.jsx:124]: This still feels brittle to me. Disregard if need be; I think that you had a good reason for keeping this the last time that I saw it.

File: src/pages/sighting/SightingCore.jsx [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+  const showCommitAlert =
+    pending && detectionComplete && allAnnotationsAssigned;
+
+  const sightingDisplayDate = get(data, ['startTime']);
+  // const encounters = get(data, ['encounters'], []);
```
Comment (Atticus29) [inline src/pages/sighting/SightingCore.jsx:129]: Intentional?

File: src/pages/sighting/SightingCore.jsx [hunk 11]
```diff
@@ -0,0 +1,279 @@
...
+        onClose={() => setHistoryOpen(false)}
+      />
+      <ConfirmDelete
+        open={deleteDialogOpen}
+        onClose={() => setDeleteDialogOpen(false)}
+        onDelete={async () => {
+          const successful = await deleteSighting(id);
+          if (successful) {
+            setDeleteDialogOpen(false);
+            history.push('/');
```
Comment (Atticus29) [inline src/pages/sighting/SightingCore.jsx:150]: Why this?

**Review by Atticus29 — APPROVED** [review]
Lots of comments and questions, but nothing deal-breaking. Nice!!

File: src/models/assetGroupSighting/useCommitAssetGroupSighting.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import axios from 'axios';
+import { useMutation } from 'react-query';
+
+export default function useCommitAssetGroupSighting() {
+  const mutation = useMutation(async agsId =>
+    axios.request({
```
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/models/assetGroupSighting/useCommitAssetGroupSighting.js:6]: it's the same

File: locale/en.json [hunk 13]
```diff
@@ -12,6 +12,9 @@
   "LABEL": "Label",
   "TYPE": "Type",
   "FOREST": "Forest",
+  "COMMIT": "Commit",
+  "SIGHTING_COMMIT_TITLE": "Ready to commit",
+  "SIGHTING_COMMIT_DESCRIPTION": "Now that all annotations are assigned to animals, this sighting can be committed. Commit this sighting to have it appear in searches and match results.",
```
[the remark this answers is not in the document]
Comment (brmscheiner) [reply locale/en.json:17]: Ahh, good question. I think eventually the button and this description will only appear if you have the appropriate permissions.

File: src/models/assetGroup/useAssetGroupSighting.js [hunk 14]
```diff
@@ -1,54 +1,29 @@
...
+  const result = useQuery(
+    getAGSQueryKey(assetGroupSightingId),
+    async () => {
+      return axios.request({
+        url: `${__houston_url__}/api/v1/asset_groups/sighting/as_sighting/${assetGroupSightingId}`,
+        method: 'get',
+      });
+    },
+    {
+      refetchInterval: 5000,
```
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/models/assetGroup/useAssetGroupSighting.js:16]: Because the sighting moves through phases based on WBIA, (detection => curation => matching). So it's very possible for the backend phase to switch while the user is on the page and we want to show them that update without refreshing.

File: src/models/sighting/useSighting.js [hunk 15]
```diff
@@ -1,52 +1,29 @@
...
+  const result = useQuery(
+    getSightingQueryKey(sightingId),
+    async () => {
+      return axios.request({
+        url: `${__houston_url__}/api/v1/sightings/${sightingId}`,
+        method: 'get',
+      });
+    },
+    {
+      refetchInterval: 5000,
```
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/models/sighting/useSighting.js:16]: Only if we think things will change without the user taking action

File: src/pages/sighting/AssetGroupSighting.jsx (hunk 3 above)
Comment (brmscheiner) [reply src/pages/sighting/AssetGroupSighting.jsx:15]: Weird it's not in the docs but I'm pretty sure I saw it on the actual object. I can double check this

File: src/pages/sighting/SightingCore.jsx (hunk 9 above)
Comment (brmscheiner) [reply src/pages/sighting/SightingCore.jsx:124]: Detection is the first stage, so if it's in any other stage it's through detection

File: src/pages/sighting/SightingCore.jsx (hunk 11 above)
Comment (brmscheiner) [reply src/pages/sighting/SightingCore.jsx:150]: Sends them back to their home page (since once the sighting is deleted the page they were on no longer exists)

File: src/pages/sighting/SightingCore.jsx (hunk 10 above)
Comment (brmscheiner) [reply src/pages/sighting/SightingCore.jsx:129]: Nope, will fix!

File: src/pages/sighting/AssetGroupSighting.jsx (hunk 3 above)
Comment (brmscheiner) [reply src/pages/sighting/AssetGroupSighting.jsx:15]: it wasn't there! however statuscode could be found on data.status so i made the adjustment. nice catch

File: src/pages/sighting/AssetGroupSighting.jsx (hunk 4 above)
Comment (brmscheiner) [reply src/pages/sighting/AssetGroupSighting.jsx:23]: Sorry i missed this one. it does the redirect, the name is just weird. check out the react-router docs for more info

File: src/pages/sighting/Sighting.jsx (hunk 5 above)
Comment (brmscheiner) [reply src/pages/sighting/Sighting.jsx:4]: Missed this one too. The structure is now different - Sighting.jsx and AssetGroupSighting.jsx do the requests and supply data to the SightingCore component which renders everything

File: src/pages/sighting/SightingCore.jsx (hunk 7 above)
Comment (brmscheiner) [reply src/pages/sighting/SightingCore.jsx:110]: good call - moved.
